common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -647,8 +945,175 @@ bool ap_wlan_hal_dwpal::sta_bss_steer(const std::string &mac, const std::string
+
+    // Start by disabling all the existent VAPs
+    for (auto &it : hostapd_config_vaps) {
+        const std::string &vap_if                    = it.first;
+        std::vector<std::string> &vap_hostapd_config = it.second;
+
+        if (!hostapd_config_set_value(vap_hostapd_config, "start_disabled", "1")) {
+            LOG(ERROR) << "Autoconfiguration: cannot disable VAP " << vap_if << ", ignoring";
+        }
+    }
+
+    // Find first VAP entry
+    auto hostapd_vap_iterator = hostapd_config_vaps.begin();
+
+    // Go through the bss_info_conf_list and change the hostapd config accordingly
+    for (const auto &bss_info_conf : bss_info_conf_list) {
+        // If a Multi-AP Agent receives an AP-Autoconfiguration WSC message containing an M2 with
+        // a Multi-AP Extension subelement with bit 4 (Tear Down bit) of the subelement value set
+        // to one (see Table 4), it shall tear down all currently operating BSS(s) on the radio
+        // indicated by the Radio Unique Identifier, and shall ignore the other attributes in the M2.
+        if ((bss_info_conf.bss_type & WSC::eWscVendorExtSubelementBssType::TEARDOWN) != 0) {
+            LOG(DEBUG) << "Autoconfiguration: received teardown, continuing";
+            continue;
+        }
+
+        auto auth_type =
+            son::wireless_utils::wsc_to_bwl_authentication(bss_info_conf.authentication_type);
+        if (auth_type == "INVALID") {
+            LOG(ERROR) << "Autoconfiguration: invalid auth_type "
+                       << int(bss_info_conf.authentication_type);
+            return false;
+        }
+        auto enc_type = son::wireless_utils::wsc_to_bwl_encryption(bss_info_conf.encryption_type);
+        if (enc_type == "INVALID") {
+            LOG(ERROR) << "Autoconfiguration: invalid enc_type "
+                       << int(bss_info_conf.encryption_type);
+            return false;
+        }
+        auto bss_type = son::wireless_utils::wsc_to_bwl_bss_type(bss_info_conf.bss_type);
+        if (bss_type == beerocks::BSS_TYPE_INVALID) {
+            LOG(ERROR) << "Autoconfiguration: invalid bss_type";
+            return false;
+        }
+        LOG(DEBUG) << "Autoconfiguration for ssid: " << bss_info_conf.ssid
+                   << " auth_type: " << auth_type << " encr_type: " << enc_type
+                   << " network_key: " << bss_info_conf.network_key << " bss_type: " << bss_type;
+
+        // We cannot configure STAs yet
+        if ((bss_info_conf.bss_type & WSC::eWscVendorExtSubelementBssType::BACKHAUL_STA) != 0) {
+            LOG(ERROR) << "Autoconfiguration: STAs are not yet supported";
+            return false;
+        }
+
+        // We only can use AP entries, skip STAs if ended up on one
+        std::string entry_mode;
+        while (hostapd_vap_iterator != hostapd_config_vaps.end() &&
+               hostapd_config_get_value(hostapd_vap_iterator->second, "mode", entry_mode) &&
+               entry_mode.compare("ap") != 0) {
```

Comment:
Doesn't `entry_mode == "ap"` work?

Reply:
Probably, I just prefer to stay away from operators overloading. If it's something that I should be doing in this code please let me know and I'll search/replace "compare()" to "==".

Reply:
Yes, please use `==` for comparing strings. It's the idiomatic way.